src/api/cvc4cpp.cpp

```diff
@@ -1731,7 +1754,7 @@ bool Term::isNull() const { return isNullHelper(); }
 bool Term::isValue() const
 {
   CVC4_API_CHECK_NOT_NULL;
-  return d_node->isConst();
+  return d_node->isConst() || isCastedReal();
```

Comment:
Again, why? Shouldn't `isConst` return true for consts independently from `isCastedReal`?

Reply:
For the same reasons I describe above.

Reply:
It seems this is not enough to satisfy the following [assertion](https://github.com/CVC4/CVC4/pull/5274/files#diff-6e779ce1b72d75a5502238049f07e8ad68d1ebb6c8f4cfc68bef2b194f1c495bR794) which fails. 
`isValue` returns true for the term `((as const (Array Real Real)) 0)`, because the node is a const, however
it returns `false` for `(store ((as const (Array Real Real)) 0) 1.0 2.0)` which is equivalent to the non-constant node `(store ((as const (Array Real Real)) 0) (to_real 1) (to_real 2))`.
I guess the same situation applies for sets (inserting casted integers into a set of reals). 

Possible suggestions I can think of to fix this. 
1- Handle this special case for the `STORE` operator, by converting `TO_REAL` to integer nodes
2- Make  `Term::isCastedReal` or `Node::isValue` recursive as `Term::isConst` in nodes.
3- Modify  `Node::isConst` to treat TO_REAL nodes as const (which Andy wants to avoid at all costs)

Reply:
After discussing this with Andy, it is became clear that having 2 different implementations of `Term::isValue` and `Term::isConst` is a bad idea. Then this question arised, does the user need `isValue` at all? So we decided to remove it until there is a convincing scenario that shows its value.

Reply:
As discussed offline, sounds good.